samples/snippets/test_get_operation.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import google.auth
+
+import pytest
+
+import get_operation
+
+
+@pytest.fixture
+def project_id():
+    _, project_id = google.auth.default()
+    return project_id
+
+
+def test_get_operation(capsys, project_id):
+    # TODO(developer): Replace this variable with your operation name.
+    operation_name = f"projects/{project_id}/locations/us-central1/operations/12345"
+
+    try:
+        operation = get_operation.get_operation(operation_name)
+        out, err = capsys.readouterr()
+        if operation.done:
+            assert "Operation is done" in out
+        else:
+            assert "Operation is in progress" in out
+    except Exception as e:
+        if "not found" not in str(e):
```

Comment:
I just tried this test.  When I get to this line, I see:

```python
print(e) # "404 Requested entity was not found."
```

As a result, the "raise" on the next line is never reached however this seems wrong to me.  A 404 error seems like something that should cause the test to fail. Am I missing something?

Reply:
In this test, there's a TODO on line 29 to replace the default arbitrary operation name with an actual operation name. The default operation name will always return "not found" because the operation ID "12345" is a fake ID. I skip "not found" errors on line 40 because the fake ID will always returns "not found". Does it make sense?

For Insights, there are quite a few steps to generate a real operation name so I didn't choose that approach. We'll have to create many resources in Insights and other services (such as BigQuery), then kick off a longrunning operation process, then poll the operation, then delete the resources, etc. IMO doing all these steps is an overkill for a get_operation unit test where we just need an operation name. Therefore I chose the approach of making up a fake operation name and adding a TODO for the developer to switch out the fake name. Let me know what you think.

Reply:
Got it. I understand the situation you are in, and your approach.  It would be better to create a fixture that can build up, and then tear down, an actual operation.

Can you please add a comment in the test, something to the effect of "We expect the following error message 404 Requested entity was not found. here, because operation 12345 does not exist."

Reply:
Done!